Firefox style ribbon

JorisGoosen opened the pull request:
So the code didn't get simplified but the UI seems clearest now
- I still like the glass look more than this, but this way at least it is glaringly obvious there are more buttons left or right
- Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1086 differently from https://github.com/jasp-stats/jasp-desktop/pull/4330

![image](https://user-images.githubusercontent.com/29062713/95210694-0f948500-07ec-11eb-95b8-559d7784ec49.png)

![image](https://user-images.githubusercontent.com/29062713/95210717-13c0a280-07ec-11eb-82b7-7889392e484b.png)

![image](https://user-images.githubusercontent.com/29062713/95210733-17ecc000-07ec-11eb-9ba2-e7336d59d60c.png)

![image](https://user-images.githubusercontent.com/29062713/95210799-263adc00-07ec-11eb-9d91-8e6cc5e77d96.png)

I'll make some nightlies as well, they will be called `firefoxRibbon`

JorisGoosen committed:
Firefox style ribbon for JASP
- So the code didn't get simplified but the UI seems clearest now
- I still like the glass look more than this, but this way at least it is glaringly obvious there are more buttons left or right
- Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1086 differently from https://github.com/jasp-stats/jasp-desktop/pull/4330

JorisGoosen commented:
Mind you, I will change the color of the tooltip separately.

TimKDJ commented:
I think this is a great improvement in terms of behaviour and very clear for the user.
Three comments which I think would make it awesome:

1. I liked the way the old menu stayed the same color and had the opacity fade on the side. I think that in combination with the arrows would be the golden ticket for me.

So the love child of these two:
<img width="1247" alt="Screenshot 2020-10-06 at 16 28 52" src="https://user-images.githubusercontent.com/18737241/95215427-3e612a00-07f1-11eb-9a5b-3825e006bd8f.png">
<img width="1250" alt="Screenshot 2020-10-06 at 16 29 13" src="https://user-images.githubusercontent.com/18737241/95215435-402aed80-07f1-11eb-9b88-ee3a25df4247.png">

2. When you press the arrow once you move 7 items over (which is quite a lot) and you cannot long press it. What about a single press is move one item over and long press keeps moving it until the user lets go.

3. When you press on the hamburger or the plus, they should stay in the pressed state. So they should remain a little smaller when activated (i.e., when you press the hamburger and then hover over it, there shouldn't be any size animation -- the same as it is for the analysis buttons).

EJWagenmakers commented:
Great improvements, and I agree with Tim's suggestions

JorisGoosen commented:
>     1. I liked the way the old menu stayed the same color and had the opacity fade on the side. I think that in combination with the arrows would be the golden ticket for me.

Yeah I liked that too, but I changed the color because it looks like the button go "under" the arrow (like in firefox I might add)

> So the love child of these two:
> <img alt="Screenshot 2020-10-06 at 16 28 52" width="1247" src="https://user-images.githubusercontent.com/18737241/95215427-3e612a00-07f1-11eb-9a5b-3825e006bd8f.png">
> <img alt="Screenshot 2020-10-06 at 16 29 13" width="1250" src="https://user-images.githubusercontent.com/18737241/95215435-402aed80-07f1-11eb-9b88-ee3a25df4247.png">
> 
>     1. When you press the arrow once you move 7 items over (which is quite a lot) and you cannot long press it. What about a single press is move one item over and long press keeps moving it until the user lets go.

I'll see what I can do. 


>     2. When you press on the hamburger or the plus, they should stay in the pressed state. So they should remain a little smaller when activated (i.e., when you press the hamburger and then hover over it, there shouldn't be any size animation -- the same as it is for the analysis buttons).

Aah yeah that is a good one, missed something there.

JorisGoosen committed:
Processed suggestions from PR
- made shadow much smaller
- buttons now either flick a bit (less than before)
- buttons can also, if pressed longer than 300ms, scroll with a constant speed
- Yes, it includes a timer :(
- Fixed the pressed state of the hamburger + plus menubutton

JorisGoosen commented:
Ok, I mostly processed your comments, I will not change the shadow color to the ui-background color though, because it really doesn't look that good imho. But making it much more narrow should help mitigate the merry-go-round effect while making sure it still looks like there is more hidden underneath.

JorisGoosen commented:
Aah, and I turned on some builds for y'all.

TimKDJ commented:
For anyone wondering, this is what it looks like now:
<img width="799" alt="Screenshot 2020-10-07 at 11 04 32" src="https://user-images.githubusercontent.com/18737241/95310634-ef68d280-088c-11eb-8230-67d78210c788.png">

Again a great improvement over the previous iteration. The scrolling also works exactly like how I would expect it to work.

When you select an analysis you get vertical bars on either side of the button. Might be good to do this also on the arrows, plus and hamburger to be consistent? (In the screenshot: new ribbon with the plus selected (top), old ribbon with frequencies selected (bottom))
<img width="801" alt="Screenshot 2020-10-07 at 11 02 27" src="https://user-images.githubusercontent.com/18737241/95311076-7027ce80-088d-11eb-8fdc-ea0a65bff118.png">

I do have to get used a little to the hamburger and plus being so far in the corner. But at this point I've looked at the JASP ribbon thousands of times so a large part of it is familiarity. Considering the ribbon is a very prominent part of the JASP application, what do others think?

EJWagenmakers commented:
"Might be good to do this also on the arrows, plus and hamburger to be consistent?" -> yes
Looks good!
E.J.

JorisGoosen commented:
Hehe, ok I'll make those buttons also show an outline on press.

JorisGoosen committed:
Give the arrows and hamburger+plus border on press

JorisGoosen commented:
Ok, looks like this now:
![image](https://user-images.githubusercontent.com/29062713/95337212-e8a08680-08b1-11eb-9dbe-71dbaef61aa4.png)

In this one the modulemenu is opened and Im pressing the rightarrow:
![image](https://user-images.githubusercontent.com/29062713/95337311-02da6480-08b2-11eb-98b8-857414ebc268.png)

![image](https://user-images.githubusercontent.com/29062713/95337401-15ed3480-08b2-11eb-93e5-470146d06a84.png)

![image](https://user-images.githubusercontent.com/29062713/95337492-2ef5e580-08b2-11eb-87ed-33a96714358a.png)

boutinb commented:
The hand cursor is not there anymore. I found it quite a good idea.

TimKDJ commented:
> In this one the modulemenu is opened and Im pressing the rightarrow:

Yes, nice, makes it a bit clearer.
I like it better in dark mode actually. Maybe because the shadow isn't so visible?

> The hand cursor is not there anymore. I found it quite a good idea.

But not the grabby hand one please, the pointy one; grabbing the bar will not be the main purpose when people interact with it (clicking analyses is).

JorisGoosen commented:
I'll just turn the grabbing off entirely, people can then either scroll vertical/horizontal (both work) or use the fancy new arrows.

And then the pointer can just be pointy on top of the buttons etc

JorisGoosen committed:
Pointing hand ftw

JorisGoosen commented:
> Yes, nice, makes it a bit clearer.
> I like it better in dark mode actually. Maybe because the shadow isn't so visible?

Well, the dark mode just looks better alround ^^

EJWagenmakers commented:
What about the bright orange bar on top of the ribbon? Is this new? Grey is easier on the eyes, and that's what I get in my version (see screenshot).
![image](https://user-images.githubusercontent.com/8603015/95375793-6bd8d100-08e0-11eb-93bb-fc7a8519b6fc.png)

JorisGoosen commented:
Yeah I just have my Windows set up in a bright orange theme, combined with dark backgrounds, it isn't part of JASP ;)
